packages/core/admin/admin/src/components/GuidedTour/index.js

```diff
@@ -0,0 +1,100 @@
+import React, { useReducer } from 'react';
+import PropTypes from 'prop-types';
+import get from 'lodash/get';
+import { GuidedTourProvider } from '@strapi/helper-plugin';
+import persistStateToLocaleStorage from './utils/persistStateToLocaleStorage';
+import arePreviousSectionsDone from './utils/arePreviousSectionsDone';
+import arePreviousStepsDone from './utils/arePreviousStepsDone';
+import reducer, { initialState } from './reducer';
+import init from './init';
+
+const GuidedTour = ({ children }) => {
+  const [{ currentStep, guidedTourState, isGuidedTourVisible, isSkipped }, dispatch] = useReducer(
+    reducer,
+    initialState,
+    init
+  );
+
+  const setCurrentStep = step => {
+    // if step is null it is intentional, we need to dispatch it
+    if (step !== null) {
+      const isStepAlreadyDone = get(guidedTourState, step);
+      const isStepToShow = arePreviousStepsDone(step, guidedTourState);
+
+      if (isStepAlreadyDone || isSkipped || !isStepToShow) {
+        return null;
+      }
+    }
+
+    persistStateToLocaleStorage.addCurrentStep(step);
```

Comment:
There can be a few reasons why localstorages fail:

- not enough storage on the device
- browser doesn't support it or the user has disabled it

Would it therefore be possible to wrap all access call to localstorage in try/ catch blocks to avoid it breaking the application?

Reply:
Interesting, I didn't know about it!
We are not sure what to do about it though. If we try/catch and it fails over and over, it would break the guided tour. Meaning that the user would constantly see the guided tour, wouldn't be able to skip it, etc - since we rely on the locale storage for the guided tour state.
WDYT @soupette?

Reply:
So, if the localstorage is a requirement for the guided tour, I think it another approach could be to check, whether you can write into the store, before creating the tour. If no localstorage, there is no tour.

To be clear: I don't think this is going to happen often, but there might a chance, so I'd rather be cautious.

Reply:
We can use the sessionStorage otherwise but if none of them are available we should use the redux store.

Reply:
I'm wondering what to do about that. To be honest I feel like lots of Strapi rely on localeStorage anyway, I tried to remove the userInfo key and my app crashed. Does that mean it would be a more general improvement to make - not only guided tour?
![image](https://user-images.githubusercontent.com/71838159/153022619-5bcb324d-0b52-49ed-a58c-faaa2ec57bd8.png)

Reply:
@ronronscelestes Alright that makes sense. In case I'm not able to even login then it doesn't make sense to introduce a safety-layer here for now. It might make sense though to create an issue somewhere to at least explore the possible impact.

Reply:
Sure that seems like a good idea, we can discuss this tomorrow while working through react lazy solution!